src/components/select/_select.scss

```diff
@@ -279,8 +279,18 @@
     pointer-events: none;
   }
 
-  &[data-invalid] ~ .#{$prefix}--select__arrow {
-    bottom: 2.5rem;
+  &[data-invalid] ~ {
+    .#{$prefix}--select__arrow {
+      bottom: 2.4rem;
+    }
+
+    .#{$prefix}--select__invalid {
+      position: absolute;
+      right: rem(34px);
+      bottom: 2.15rem;
```

Comment:
Where does `2.15rem` come from?

Reply:
it's because we are relying on absolute positioning to position the icon. I think I would prefer to just rewrite the component structure instead to avoid absolute positioning like I did for https://github.com/IBM/carbon-components/pull/1622